### PR TITLE
VIH-8035 VHO saves booking with Staff Member added

### PR DESCRIPTION
**Pull request by alberto-hortiguela**
Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/VIH-8035

### Change description ###
Staff Member participant to be added with 'Staff Member' AAD Group when hearing is booked.
Unit test update

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: AdminWebsite/AdminWebsite/Services/UserAccountService.cs [hunk 1]
```diff
@@ -280,6 +281,9 @@ public async Task AssignParticipantToGroup(string username, string userRole)
                 case JohRole:
```
Comment (kevin-hmcts) [inline AdminWebsite/AdminWebsite/Services/UserAccountService.cs:281]: Is it worth checking the two parameters coming in and also a default for the case block?

File: AdminWebsite/AdminWebsite.UnitTests/Services/UserAccountServiceTests.cs [hunk 2]
```diff
@@ -111,6 +116,21 @@ public async Task Should_add_JOH_role_to_JOH_user_group()
                 Times.Once);
         }
 
+        [Test]
+        public async Task Should_add_StaffMember_role_to_Staff_Member_user_group()
```
Comment (kevin-hmcts) [inline AdminWebsite/AdminWebsite.UnitTests/Services/UserAccountServiceTests.cs:120]: Consider using Pascal case for test method names, eg:
ShouldAddStaffMemberRoleToStaffMemberUserGroup, is maybe a  little less cognitive strain for the reviewer. 

Also consider: 
NameOfMethodUnderTest_AddStaffMemberRoleToStaffMemberUserGroup to add some context.

File: AdminWebsite/AdminWebsite.UnitTests/Services/UserAccountServiceTests.cs [hunk 3]
```diff
@@ -111,6 +116,21 @@ public async Task Should_add_JOH_role_to_JOH_user_group()
...
+        {
+            await _service.AssignParticipantToGroup("staff@hmcts.net", "Staff Member");
+
+            _userApiClient.Verify(
+                x => x.AddUserToGroupAsync(
+                    It.Is<AddUserToGroupRequest>(y => y.GroupName == UserAccountService.External)),
```
Comment (PrasannaDommalapati) [inline AdminWebsite/AdminWebsite.UnitTests/Services/UserAccountServiceTests.cs:126]: @alberto-hortiguela it should be Internal Group for Staff members

File: AdminWebsite/AdminWebsite/Services/UserAccountService.cs (hunk 1 above)
Comment (alberto-hortiguela) [reply AdminWebsite/AdminWebsite/Services/UserAccountService.cs:281]: We have unit test to check the userName that throws exceptions if this is null> Should_throw_exception_when_add_user_to_group_has_null_user_name()

The userRole can be anything and it does not need to meet the case block ( no default needed )

File: AdminWebsite/AdminWebsite.UnitTests/Services/UserAccountServiceTests.cs (hunk 2 above)
Comment (alberto-hortiguela) [reply AdminWebsite/AdminWebsite.UnitTests/Services/UserAccountServiceTests.cs:120]: That´s done now

File: AdminWebsite/AdminWebsite.UnitTests/Services/UserAccountServiceTests.cs (hunk 3 above)
Comment (alberto-hortiguela) [reply AdminWebsite/AdminWebsite.UnitTests/Services/UserAccountServiceTests.cs]: That's done now

**Review by MichaelKeaneathmcts — APPROVED** [review]

File: AdminWebsite/AdminWebsite/Services/UserAccountService.cs [hunk 4]
```diff
@@ -269,17 +271,24 @@ public async Task DeleteParticipantAccountAsync(string username)
...
-            
+        {   
             switch (userRole)
             {
                 case RepresentativeRole:
+                    await AddGroup(username, External);
                     await AddGroup(username, VirtualRoomProfessionalUser);
                     break;
                 case JohRole:
+                    await AddGroup(username, External);
```
Comment (PrasannaDommalapati) [inline AdminWebsite/AdminWebsite/Services/UserAccountService.cs:282]: is it worth considering to use userRole=="Staff Member" ? await AddGroup(username, Internal) : await AddGroup(username, External)

Comment (alberto-hortiguela) [reply AdminWebsite/AdminWebsite/Services/UserAccountService.cs:282]: I believe that would work however the switch case reflects perfectly the treatment for the StaffMember, Representative and JohRole...the default will be always external

**Review by PrasannaDommalapati — APPROVED** [review]
LGTM

**Pull request merged**